docs/user_guide/configuring.rst

```diff
@@ -277,6 +277,93 @@ at the bottom. You can hide these buttons with the following configuration:
    }
 
 
+Add a dropdown to switch between docs versions
+==============================================
+
+You can add a button to your site's navbar or sidebars that allows users to
+switch between versions of the documentation. The versions shown in the
+dropdown are populated from a JSON file containing a list of mappings, where
+each mapping contains a key ``"version"`` mapped to a string value representing
+the version number. Each mapping may optionally contain a key ``"name"`` mapped
+to a name for that version (e.g., "latest", "stable", "dev", etc). Here is an
+example:
+
+.. code:: json
+
+    [
+        {
+            "name": "v2.1 (stable)",
+            "version": "2.1"
+        },
+        {
+            "version": "2.0"
+        },
+        {
+            "version": "1.1"
+        },
+    ]
+
+The location of this JSON file should be included in ``conf.py`` as the
+``switcher_json_url`` value within the ``html_context`` dictionary.
+Additionally, you must provide a *template URL* that will yield the correct
+address for the doc versions once the version string is inserted into the
+template in place of the ``{version}`` placeholder. Finally, specify in
+``switcher_version_match`` a value that the version entries in the JSON file
+will be matched against (typically this will be the sphinx variable ``version``
+or ``release`` depending on how granular your docs versioning is; see
+`the Sphinx documentation <https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information>`__
+for more info). An example configuration would be:
+
+.. code:: python
```

Comment:
```suggestion

Configure Sphinx to use this file to populate the switcher
----------------------------------------------------------

You must next configure your documentation to use this file, and to properly create URLs for other documentation versions using the variables stored within it.

To do so, you must add the following configuration variables to the ``html_context`` key in your ``conf.py`` file:

- ``switcher_json_url``:  The **persistent** URL of the switcher configuration that serves as the "source of truth". Each version of your documentation should point to the same value, so that they all have the same database of switcher versions to use. This should be a fully-resolved URL, like ``http://mydocumentation.readthedocs.org/en/latest/_static/switcher.json``.

  .. note::

     As a general rule, a good choice is to use a URL that is always associated with the most 
     recent documentation build (as above, where it points to a location in the build
     tree of version "latest").  Alternatively the JSON could be hosted in a persistent place 
     outside of any one documentation build (e.g., ``https://mysite.org/switcher.json``).
- ``switcher_template_url``: A *template URL* that has a field where the version value should be inserted. This is used to create the URL that the switcher links will point to. For example, if your ``switcher_template_url`` is ``"https://mysite.org/en/version-{version}/"`` then the switcher link for version ``1.1`` will be ``"https://mysite.org/en/version-1.1/"``
- ``switcher_version_match``: A string that defines the version of the *current* documentation. This is used to tell the switcher which version of the documentation is currently open. It is best if you make this value match the ``version`` or ``release`` value of your package, by automatically loading it as a variable from your package (see the example below, and see
`the Sphinx documentation <https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information>`__ for more info).

Below is an example configuration that automatically loads the version from a package, and defines a switcher URL template that follows the ``ReadTheDocs`` pattern.

.. code:: python
```

A few more additions to try and clarify each piece more explicitly